gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/TableInfo.java

```diff
@@ -0,0 +1,741 @@
+      this.tableId = tableInfo.tableId;
+      this.friendlyName = tableInfo.friendlyName;
+      this.description = tableInfo.description;
+      this.schema = tableInfo.schema;
+      this.numBytes = tableInfo.numBytes;
+      this.numRows = tableInfo.numRows;
+      this.creationTime = tableInfo.creationTime;
+      this.expirationTime = tableInfo.expirationTime;
+      this.lastModifiedTime = tableInfo.lastModifiedTime;
+      this.viewQuery = tableInfo.viewQuery;
+      this.userDefinedFunctions = tableInfo.userDefinedFunctions;
+      this.type = tableInfo.type;
+      this.externalConfiguration = tableInfo.externalConfiguration;
+      this.location = tableInfo.location;
+      this.streamingBuffer = tableInfo.streamingBuffer;
+    }
+
+    Builder creationTime(Long creationTime) {
+      this.creationTime = creationTime;
+      return this;
+    }
+
+    /**
+     * Sets a user-friendly description for the table.
+     */
+    public Builder description(String description) {
+      this.description = firstNonNull(description, Data.<String>nullOf(String.class));
+      return this;
+    }
+
+    Builder etag(String etag) {
+      this.etag = etag;
+      return this;
+    }
+
+    /**
+     * Sets the time when this table expires, in milliseconds since the epoch. If not present, the
+     * table will persist indefinitely. Expired tables will be deleted and their storage reclaimed.
+     */
+    public Builder expirationTime(Long expirationTime) {
+      this.expirationTime = firstNonNull(expirationTime, Data.<Long>nullOf(Long.class));
+      return this;
+    }
+
+    /**
+     * Sets the data format, location, and other properties of a table stored outside of BigQuery.
+     * This property is experimental and might be subject to change or removed.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/federated-data-sources">Federated Data
+     *     Sources</a>
+     */
+    public Builder externalConfiguration(ExternalDataConfiguration externalConfiguration) {
+      this.externalConfiguration = externalConfiguration;
+      return this;
+    }
+
+    /**
+     * Sets a user-friendly name for the dataset.
+     */
+    public Builder friendlyName(String friendlyName) {
+      this.friendlyName = firstNonNull(friendlyName, Data.<String>nullOf(String.class));
+      return this;
+    }
+
+    Builder id(String id) {
+      this.id = id;
+      return this;
+    }
+
+    Builder lastModifiedTime(Long lastModifiedTime) {
+      this.lastModifiedTime = lastModifiedTime;
+      return this;
+    }
+
+    Builder location(String location) {
+      this.location = location;
+      return this;
+    }
+
+    Builder numBytes(Long numBytes) {
+      this.numBytes = numBytes;
+      return this;
+    }
+
+    Builder numRows(Long numRows) {
+      this.numRows = numRows;
+      return this;
+    }
+
+    /**
+     * Sets the table's schema. Providing a schema is not necessary when {@link #viewQuery} is
```

Comment:
I assume that this is also not necessary for EXTERNAL (when ExternalDataConfiguration is provided)?